providers/common/ciphers/ciphers_locl.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/aes.h>
+#include <openssl/modes.h>
+
+typedef struct prov_aes_cipher_st PROV_AES_CIPHER;
+
+typedef struct prov_aes_key_st {
+    union {
+        double align;
```

Comment:
Consider an `intmax_t align2` it could have worse alignment restrictions.

Reply:
I'm somewhat reluctant to make that change in this PR. This is how it is done in the original code and there are enough changes as it is in this PR without fiddling around with alignment etc. Could be handled by a separate PR possibly.

Reply:
A separate PR that tries to catch all of them seems sensible.

Reply:
I've raised an issue #8791 about this.